src/backend/session.rs

```diff
@@ -0,0 +1,193 @@
+            let _ = monitor.session_closed(handle_token).await;
+        }
+        server.remove::<Self, _>(&self.path).await?;
+        Ok(())
+    }
+
+    #[zbus(signal)]
+    async fn closed(signal_emitter: &SignalEmitter<'_>) -> zbus::Result<()>;
+}
+
+impl Debug for Session {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Session").field("path", &self.path).finish()
+    }
+}
+
+#[async_trait]
+/// A trait that backends that create long-lived sessions should implement
+/// to be notified when a session has been closed.
+pub trait SessionImpl: Send + Sync {
+    async fn session_closed(&self, session_handle_token: HandleToken)
+        -> crate::backend::Result<()>;
+}
+
+#[derive(SerializeDict, Type, Debug)]
+#[zvariant(signature = "dict")]
+pub struct CreateSessionResponse {
+    session_id: HandleToken,
+}
+
+impl CreateSessionResponse {
+    pub fn new(handle_token: HandleToken) -> Self {
+        Self {
+            session_id: handle_token,
+        }
+    }
+}
+
+#[derive(Default)]
+pub(crate) struct SessionManager {
+    sessions: Arc<Mutex<HashMap<HandleToken, Session>>>,
+}
+
+impl SessionManager {
+    /// Tests whether a session with the given handle token already exists.
+    pub fn contains(&self, handle_token: &HandleToken) -> bool {
+        #[cfg(feature = "tracing")]
+        tracing::debug!("SessionManager::contains: tracked sessions: {}", &self);
+        let sessions = self.sessions.lock().unwrap();
+        sessions.contains_key(handle_token)
+    }
+
+    /// Add a session to the list of tracked sessions.
+    pub fn add(&mut self, session: Session) -> crate::backend::Result<()> {
+        let mut sessions = self.sessions.lock().unwrap();
+        let handle_token = session.handle_token();
+        if sessions.insert(handle_token.clone(), session).is_some() {
+            Err(self.error_exist(&handle_token))
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Remove a session from the list of tracked sessions.
+    pub fn remove(&mut self, handle_token: &HandleToken) -> crate::backend::Result<()> {
+        let mut sessions = self.sessions.lock().unwrap();
+        if sessions.remove(handle_token).is_some() {
+            Ok(())
+        } else {
+            Err(self.error_notfound(handle_token))
+        }
+    }
+
+    pub fn error_exist(&self, handle_token: &HandleToken) -> PortalError {
+        let errormsg = format!("A session with handle `{handle_token}` already exists");
+        #[cfg(feature = "tracing")]
+        tracing::error!("{}", errormsg.as_str());
+        PortalError::Exist(errormsg)
+    }
+
+    pub fn error_notfound(&self, handle_token: &HandleToken) -> PortalError {
+        let errormsg = format!("Unknown session: `{handle_token}`");
+        #[cfg(feature = "tracing")]
+        tracing::error!("{}", errormsg.as_str());
+        PortalError::NotFound(errormsg)
+    }
+}
+
+impl Display for SessionManager {
```

Comment:
No need for this, please drop it

Reply:
It is needed for the debug log in `SessionManager::contains()`:
```rust
pub fn contains(&self, handle_token: &HandleToken) -> bool {
    #[cfg(feature = "tracing")]
    tracing::debug!("SessionManager::contains: tracked sessions: {}", &self);
    self.sessions.contains_key(handle_token)
}
```
For a debug output that looks like this:
```
DEBUG ashpd::backend::session: SessionManager::contains: tracked sessions: [`foo`, `baz`, `bar`]
```

Reply:
Impl Debug then, not Display.

Reply:
Right, the debug output for contains will display a list that looks like this now:
```
[HandleToken("foo"), HandleToken("baz"), HandleToken("bar")]
```
which is less concise, but I guess is fine given that this is debug output anyway.
I'm getting rid of the additional itertools dependency by the way, which was required by the use of the [`format_with` method](https://docs.rs/itertools/0.12.1/itertools/trait.Itertools.html#method.format_with).

Reply:
that is good enough :) it is just a debug output